include/volume/volume_cooling_gaussians_crhmc.hpp

```diff
@@ -0,0 +1,550 @@
+#ifndef VOLUME_COOLING_GAUSSIANS_HPP
+#define VOLUME_COOLING_GAUSSIANS_HPP
+
+//#define VOLESTI_DEBUG
+
+#include <iterator>
+#include <vector>
+#include <list>
+#include <math.h>
+#include <chrono>
+
+#include "cartesian_geom/cartesian_kernel.h"
+#include "random_walks/gaussian_helpers.hpp"
+#include "random_walks/gaussian_ball_walk.hpp"
+#include "random_walks/gaussian_cdhr_walk.hpp"
+#include "sampling/random_point_generators.hpp"
+#include "volume/math_helpers.hpp"
+#include "random_walks/random_walks.hpp"
+
+//new include for crhmc
+#include "Eigen/Eigen"
+#include "cartesian_geom/cartesian_kernel.h"
+#include "volume/sampling_policies.hpp"
+#include "ode_solvers/ode_solvers.hpp"
+#include "preprocess/crhmc/crhmc_input.h"
+#include "preprocess/crhmc/crhmc_problem.h"
+#include "sampling/random_point_generators.hpp"
+#include "sampling/sampling.hpp"
+#include "random.hpp"
+#include <vector>
+#include "random_walks/random_walks.hpp"
+#include "generators/known_polytope_generators.h"
+
+
+// define types
+using NT = double;
+using Kernel = Cartesian<NT>;
+using Point = typename Kernel::Point;
+using Func = GaussianFunctor::FunctionFunctor<Point>;
+using Grad = GaussianFunctor::GradientFunctor<Point>;
+using Hess = GaussianFunctor::HessianFunctor<Point>;
+
+using NegativeLogprobFunctor = GaussianFunctor::FunctionFunctor<Point>; //Func
+using NegativeGradientFunctor = GaussianFunctor::GradientFunctor<Point>; //Grad
+using HessianFunctor = GaussianFunctor::HessianFunctor<Point>; //Hess
+
+using PolytopeType = HPolytope<Point>;
+using MT = PolytopeType::MT;
+using VT = Eigen::Matrix<NT, Eigen::Dynamic, 1>;
+using func_params = GaussianFunctor::parameters<NT, Point>;
+using RNG = BoostRandomNumberGenerator<boost::mt19937, NT>;
+
+//Param building -> see sampling_functions.cpp
+using Input = crhmc_input<MT, Point, Func, Grad, Hess>;
+using CrhmcProblem = crhmc_problem<Point, Input>;
+
+////////////////////////////// Algorithms
+
+// Gaussian Anealling
+
+// Implementation is based on algorithm from paper "A practical volume algorithm",
+// Springer-Verlag Berlin Heidelberg and The Mathematical Programming Society 2015
+// Ben Cousins, Santosh Vempala
+
+// Compute the first variance a_0 for the starting gaussian
+template <typename Polytope, typename NT>
+void get_first_gaussian(Polytope& P,
```

Comment:
Has this function differences from the `get_first_gaussian` in `volume/volume_cooling_gaussian.hpp`?